### PR TITLE
Slider

### PR DESCRIPTION
**Pull request by cfumiwu**
Creating a slider that can control over the visualizer timeline as well as the popup windows timeline and a timestamp in the main window.

### CODE REVIEW

File: src/main/java/org/openjdk/shenandoah/ToolbarPanel.java [hunk 1]
```diff
@@ -191,6 +232,7 @@ private void setEnablePlaybackButtons(boolean b) {
         playPauseButton.setEnabled(b);
```
Comment (earthling-amzn) [inline src/main/java/org/openjdk/shenandoah/ToolbarPanel.java:232]: Should the slider similarly be enabled/disabled for playback and _live_ mode?

File: src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java [hunk 2]
```diff
@@ -181,6 +190,17 @@ public void actionPerformed(ActionEvent ae) {
 
                     f[0] = changeScheduleInterval(1, service, f[0], renderRunner);
                 }
+
+                int lastSnapshotIndex = totalSnapshotSize - 1;
+                toolbarPanel.setEndSnapshotButtonListener(new ActionListener() {
```
Comment (earthling-amzn) [inline src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java]: You could use a lambda for these `ActionListeners`

**Review by earthling-amzn — COMMENTED** [review]
Looks good, but not clear how the slider behaves when visualizer is in _live_ mode? Is it disabled?

File: src/main/java/org/openjdk/shenandoah/ToolbarPanel.java (hunk 1 above)
Comment (cfumiwu) [reply src/main/java/org/openjdk/shenandoah/ToolbarPanel.java:232]: In the live mode, you can still move the slider but it won't create any effect. Is it better to disable it? Or leaving it this way is fine?

Comment (earthling-amzn) [reply src/main/java/org/openjdk/shenandoah/ToolbarPanel.java:232]: Probably better to disable it. Having it usable but with no effect would be confounding for users.

Comment (cfumiwu) [reply src/main/java/org/openjdk/shenandoah/ToolbarPanel.java:232]: This should be fixed in the latest commit.

Comment (earthling-amzn) [reply src/main/java/org/openjdk/shenandoah/ToolbarPanel.java:232]: Thank you!

**Review by earthling-amzn — APPROVED** [review]

**Pull request merged**